x-pack/test/security_solution_cypress/cypress/e2e/detection_response/rule_management/related_integrations/related_integrations.cy.ts

```diff
@@ -221,14 +221,29 @@ describe('Related integrations', { tags: ['@ess', '@brokenInServerless'] }, () =
         openTable();
         filterBy(RELATED_INTEGRATION_FIELD);
 
-        RULE_RELATED_INTEGRATIONS.forEach((integration) => {
-          cy.contains(
-            FIELD(RELATED_INTEGRATION_FIELD),
-            `{"package":"${integration.package}"${
-              integration.integration ? `,"integration":"${integration.integration}"` : ''
-            },"version":"${integration.version}"}`
-          );
-        });
+        cy.get(FIELD(RELATED_INTEGRATION_FIELD))
+          .invoke('text')
+          .then((stringValue) => {
+            // Integrations are displayed in the flyout as a string with a format like so:
+            // '{"package":"aws","version":"1.17.0","integration":"unknown"}{"package":"mock","version":"1.1.0"}{"package":"system","version":"1.17.0"}'
+            // We need to parse it to an array of valid objects before we can compare it to the expected value
+            // Otherwise, the test might fail because of the order of the properties in the objects in the string
+            const jsonStringArray = stringValue.split('}{');
+
+            const validJsonStringArray = createValidJsonStringArray(jsonStringArray);
+
+            const parsedIntegrations = validJsonStringArray.map((jsonString) =>
+              JSON.parse(jsonString)
+            );
+
+            RULE_RELATED_INTEGRATIONS.forEach((integration) => {
+              expect(parsedIntegrations).to.deep.include({
+                package: integration.package,
+                version: integration.version,
+                ...(integration.integration ? { integration: integration.integration } : {}),
+              });
+            });
+          });
```

Comment:
Why are we testing the Alert Details flyout in the first place? This is not a part of the related integrations feature and we don't own the UI => which can be a source of flakiness by itself.

Can we just check what's in the generated alerts via the API?

Reply:
Also, it shouldn't be a Cypress tests, because we could check what's in the generated alerts in an integration test.